### PR TITLE
Added False Position algorithm

### PR DESCRIPTION
**Pull request by MatteoRaso**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: arithmetic_analysis/false_position.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    INPUTS:
+    f: The function which who's root we want to find.
+    lower_bound: A value at which f(lower_bound) < 0
+    upper_bound: A value at which f(upper_bound) > 0
+    error: The maximum tolerated error
+    
+    OUTPUT:
+    The approximated root bounded by lower_bound and upper_bound
+    """
+    if f(lower_bound) > 0:
```
Comment (onlinejudge95) [inline arithmetic_analysis/false_position.py]: For edge case like these, how about raising an exception

File: arithmetic_analysis/false_position.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    
+    INPUTS:
+    f: The function which who's root we want to find.
+    lower_bound: A value at which f(lower_bound) < 0
+    upper_bound: A value at which f(upper_bound) > 0
+    error: The maximum tolerated error
+    
+    OUTPUT:
+    The approximated root bounded by lower_bound and upper_bound
+    """
```
Comment (onlinejudge95) [inline arithmetic_analysis/false_position.py]: Some doctests would help

File: arithmetic_analysis/false_position.py (hunk 1 above)
Comment (MatteoRaso) [reply arithmetic_analysis/false_position.py]: I'm not sure if I understand what you mean. The program would still run without that check, but it wouldn't be guaranteed to converge.

Comment (cclauss) [reply arithmetic_analysis/false_position.py]:
```python
if f(lower_bound) > 0:
    raise ValueError("f(lower_bound) must be less than zero")

Comment (MatteoRaso) [reply arithmetic_analysis/false_position.py]: Thank you for the clarification.

File: arithmetic_analysis/false_position.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                + upper_bound
+            )
+            # Narrowing the interval.
+            if f(root) > 0:
+                upper_bound = root
+
+            else:
+                lower_bound = root
+
+        return root
```
Comment (TheSuperNoob) [inline arithmetic_analysis/false_position.py]: You should round this to the same number of decimals as your error tolerance as the last few digits will not be accurate. 

Also consider using pythons decimal module for arbitrary precision if you want the error tolerance to be arbitrarily small.

**Pull request closed without merging**